src/Autofac.Integration.AspNetCore.Multitenant/MultitenantRequestServicesMiddleware.cs

```diff
@@ -56,7 +70,8 @@ public async Task Invoke(HttpContext context)
             IServiceProvidersFeature existingFeature = null!;
             try
             {
-                var autofacFeature = RequestServicesFeatureFactory.CreateFeature(context, _multitenantContainer.Resolve<IServiceScopeFactory>());
+                var factory = _serviceScopeFactory.Invoke(_multitenantContainer.GetCurrentTenantScope());
+                var autofacFeature = RequestServicesFeatureFactory.CreateFeature(context, factory);
```

Comment:
It'd still be interesting to see if we even need this `RequestServicesFeatureFactory` now that there's no .NET 2.1.

Reply:
Would you want to remove the entire procedure to build the expression or only the bits with switching based on parameter count?

Reply:
Honestly, if we can skip the whole feature and just new-up the thing and have it work, let's do that. There's not a lot of value in having all this extra expression building to wrap a simple `new` operation.